libsolidity/lsp/FileRepository.cpp

```diff
@@ -69,6 +91,52 @@ void FileRepository::setSourceByUri(string const& _uri, string _source)
 	m_sourceCodes[sourceUnitName] = std::move(_source);
 }
 
+Result<boost::filesystem::path> FileRepository::tryResolvePath(std::string const& _strippedSourceUnitName) const
+{
+	if (
+		boost::filesystem::path(_strippedSourceUnitName).has_root_path() &&
+		boost::filesystem::exists(_strippedSourceUnitName)
+	)
+		return boost::filesystem::path(_strippedSourceUnitName);
+
+	vector<boost::filesystem::path> candidates;
+	vector<reference_wrapper<boost::filesystem::path const>> prefixes = {m_basePath};
+	prefixes += (m_includePaths | ranges::to<vector<reference_wrapper<boost::filesystem::path const>>>);
+	auto const defaultInclude = m_basePath / "node_modules";
```

Comment:
This looks like a better place for adding special paths than `sourceUnitNameToUri()` but this still makes it impossible not to have any include paths at all - and this is a very common case. Maybe a more logical place to add this would be `changeConfiguration()`? That function can discern between "include paths not set" (i.e. `null`) and "include paths set is empty" (i.e. `[]`).

Unless the LSP clients do not care and would just always send `[]`... What do you think?

Reply:
Yes, it's going to add that default include only if the user hasn't supplied any. Do you explicitly want to support dropping the default include and have zero other include paths set? I think this case is rather artificial. The default include path is for ease-of-use, as that's most likely the most often used include path being used in the wild (AFAICT). If people chose to roll their own, they can do, which would not auto-add the default include path. Hmmm....

Reply:
Yeah, I think it would be best if you could drop the default path from the set. I mean, it probably won't hurt anyone in a vast majority of cases but it just does not seem quite right. If the client has a way to say explicitly "no extra paths" (and this depends on client's UI), the server should have an option to respect that. In some corner cases you could have a project that does not use `node_modules` for contracts but still have a stray dir there because you were installing some extra tools via npm.

Now that I think of it - is there a mechanism for the server to provide the "default" settings? Because the "right" way to solve this IMO would be to just put `node_modules` path in these defaults so that user does not have to do anything but is also able to modify them.

Overall this is not a deal-breaker and I'm going to accept the PR regardless of this but I'm just thinking that maybe we could solve this better.

Reply:
Marking as resolved.

We discussed it on the call and looks like there is not way for the client to query the configuration. So my idea is not viable after all.